### PR TITLE
Review usage of string.Substring (part 1)

### PR DESCRIPTION
**Pull request by Bond-009**
Reduced allocations by replacing string.Substring with ReadOnlySpan<char>.Slice

### CODE REVIEW

File: Emby.Server.Implementations/Library/MediaSourceManager.cs [hunk 1]
```diff
@@ -881,6 +881,7 @@ public async Task CloseLiveStream(string id)
         public void Dispose()
         {
             Dispose(true);
+            GC.SuppressFinalize(this);
```
Comment (cvium) [inline Emby.Server.Implementations/Library/MediaSourceManager.cs:884]: Seems unrelated

File: Emby.Server.Implementations/LiveTv/TunerHosts/M3uParser.cs [hunk 2]
```diff
@@ -158,15 +158,14 @@ private ChannelInfo GetChannelnfo(string extInf, string tunerHostId, string medi
         private string GetChannelNumber(string extInf, Dictionary<string, string> attributes, string mediaUrl)
         {
             var nameParts = extInf.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
-            var nameInExtInf = nameParts.Length > 1 ? nameParts[nameParts.Length - 1].Trim() : null;
+            var nameInExtInf = nameParts.Length > 1 ? nameParts[^1].AsSpan().Trim() : ReadOnlySpan<char>.Empty;
```
Comment (cvium) [inline Emby.Server.Implementations/LiveTv/TunerHosts/M3uParser.cs:161]: `^` is new to me. Guess I should read the updates more in-depth.

File: Emby.Server.Implementations/Library/MediaSourceManager.cs (hunk 1 above)
Comment (Bond-009) [reply Emby.Server.Implementations/Library/MediaSourceManager.cs:884]: Yep... the analyzers made me do it :p

File: Emby.Server.Implementations/Services/ServiceExec.cs [hunk 3]
```diff
@@ -105,7 +106,12 @@ public static Task<object> Execute(Type serviceType, IRequest request, object in
             }
 
             var expectedMethodName = actionName.Substring(0, 1) + actionName.Substring(1).ToLowerInvariant();
-            throw new NotImplementedException(string.Format("Could not find method named {1}({0}) or Any({0}) on Service {2}", requestDto.GetType().GetMethodName(), expectedMethodName, serviceType.GetMethodName()));
+            throw new NotImplementedException(
+                string.Format(
+                    CultureInfo.InvariantCulture,
+                    "Could not find method named {1}({0}) or Any({0}) on Service {2}",
+                    requestDto.GetType().GetMethodName(),
+                    expectedMethodName, serviceType.GetMethodName()));
```
Comment (cvium) [inline Emby.Server.Implementations/Services/ServiceExec.cs:114]: YIKES
```suggestion
                    expectedMethodName,
                    serviceType.GetMethodName()));
```

File: Emby.Server.Implementations/LiveTv/TunerHosts/M3uParser.cs (hunk 2 above)
Comment (Bond-009) [reply Emby.Server.Implementations/LiveTv/TunerHosts/M3uParser.cs:161]: C#8 feature IIRC, it's amazing

File: Emby.Server.Implementations/Services/ServiceHandler.cs [hunk 4]
```diff
@@ -55,15 +55,18 @@ public static string GetSanitizedPathInfo(string pathInfo, out string contentTyp
...
+        private static string GetFormatContentType(ReadOnlySpan<char> format)
         {
-            // built-in formats
-            switch (format)
+            if (format.Equals("json", StringComparison.Ordinal))
             {
-                case "json": return "application/json";
-                case "xml": return "application/xml";
-                default: return null;
+                return "application/json";
```
Comment (cvium) [inline Emby.Server.Implementations/Services/ServiceHandler.cs:62]: Should this be `MediaTypeNames.Application.Json` instead?

File: Emby.Server.Implementations/Services/ServiceHandler.cs [hunk 5]
```diff
@@ -55,15 +55,18 @@ public static string GetSanitizedPathInfo(string pathInfo, out string contentTyp
...
             }
+            else if (format.Equals("xml", StringComparison.Ordinal))
+            {
+                return "application/xml";
```
Comment (cvium) [inline Emby.Server.Implementations/Services/ServiceHandler.cs:66]: Same as above
